Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/Analytics.kt

```diff
@@ -1,34 +1,156 @@
 package de.rki.coronawarnapp.datadonation.analytics
 
+import androidx.annotation.VisibleForTesting
 import de.rki.coronawarnapp.appconfig.AppConfigProvider
 import de.rki.coronawarnapp.datadonation.analytics.modules.DonorModule
+import de.rki.coronawarnapp.datadonation.analytics.server.DataDonationAnalyticsServer
+import de.rki.coronawarnapp.datadonation.analytics.storage.AnalyticsSettings
+import de.rki.coronawarnapp.datadonation.analytics.storage.LastAnalyticsSubmissionLogger
 import de.rki.coronawarnapp.datadonation.safetynet.DeviceAttestation
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaDataRequestAndroid
+import de.rki.coronawarnapp.storage.LocalData
+import org.joda.time.Hours
+import org.joda.time.Instant
+import timber.log.Timber
 import javax.inject.Inject
 import javax.inject.Singleton
+import kotlin.random.Random
 
 @Singleton
 class Analytics @Inject constructor(
+    private val dataDonationAnalyticsServer: DataDonationAnalyticsServer,
     private val appConfigProvider: AppConfigProvider,
     private val deviceAttestation: DeviceAttestation,
-    private val donorModules: Set<DonorModule>,
-    private val settings: AnalyticsSettings
+    // @JvmSuppressWildcards is needed for @IntoSet injection in Kotlin
+    private val donorModules: Set<@JvmSuppressWildcards DonorModule>,
+    private val settings: AnalyticsSettings,
+    private val logger: LastAnalyticsSubmissionLogger
 ) {
+    private suspend fun trySubmission(ppaData: PpaData.PPADataAndroid): Boolean {
+        try {
+            val ppaAttestationRequest = PPADeviceAttestationRequest(
+                ppaData = ppaData
+            )
 
-    val isEnabled: Boolean = true
+            val attestation = deviceAttestation.attest(ppaAttestationRequest)
```

Comment:
This only means that SafetyNet passed "generally". The analytics config contains specific requirements that you need to check, i.e.

`attestation.requirePass(analyticsConfig.requirements)`